### DISCUSSION TITLE
"Body with Multiple Examples" only showing single example v0.100.1

### DISCUSSION THREAD

**Discussion by Cozmo25** [fastapi/fastapi#9958] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

From the Fastapi docs: https://fastapi.tiangolo.com/tutorial/schema-extra-example/?h=swagg#body-with-multiple-examples

```python


from typing import Annotated

from fastapi import Body, FastAPI
from pydantic import BaseModel

app = FastAPI()


class Item(BaseModel):
    name: str
    description: str | None = None
    price: float
    tax: float | None = None


@app.put("/items/{item_id}")
async def update_item(
    *,
    item_id: int,
    item: Annotated[
        Item,
        Body(
            examples=[
                {
                    "name": "Foo",
                    "description": "A very nice Item",
                    "price": 35.4,
                    "tax": 3.2,
                },
                {
                    "name": "Bar",
                    "price": "35.4",
                },
                {
                    "name": "Baz",
                    "price": "thirty five point four",
                },
            ],
        ),
    ],
):
    results = {"item_id": item_id, "item": item}
    return results
```

### Description

The above code only shows a single example for the relevant path in /docs using Fastapi v0.100.1 and Pydantic 2.1.1.

![Screenshot from 2023-07-28 16-18-56](https://github.com/tiangolo/fastapi/assets/12619962/c77d089f-719f-4d82-8000-768681e2d541)

However, if I roll back to version 0.98.0 and Pydantic 1 it works fine with a drop-down to select each example.

![Screenshot from 2023-07-28 16-21-18](https://github.com/tiangolo/fastapi/assets/12619962/0a7778ec-953a-438c-a4cd-1209611173c7)

Also, the docs show named examples in the /docs drop-down, but it is not clear how to achieve that using the methods shown in the docs.

Apologies if I have not correctly managed the migration to Pydantic 2, but I would expect that the example in the docs should work.

### Operating System

Linux

### FastAPI Version

0.100.1

### Python Version

3.11.4

**Comment by arjwilliams** [comment, marked as answer]
This is a duplicate of #9794.

@Cozmo25 - Can you check out that discussion, and possibly whether the PR #9866 fixes the issue.

**Reply by tiangolo** [reply]
Yep, this is a duplicate. 🤓 

In any case, @Cozmo25, there's a new parameter `openapi_examples` just for this, to allow declaring examples that show up in Swagger UI while it implements support for multiple JSON Schema examples. It's available now in FastAPI 0.103.0, the new docs are here: https://fastapi.tiangolo.com/tutorial/schema-extra-example/#openapi-specific-examples 🎉

**Reply by Cozmo25** [reply]
That’s great thank you @tiangolo!

**Comment by Cozmo25** [comment]
Hi @arjwilliams - apologies for the duplicate, I tried searching for this issue but my search skills clearly suck 😂 

#9866 would fix the issue for me and offer the option of passing a dict of examples in the body which is the method I’ve been using up to v0.98.0 and works well with the interactive docs that are used by my colleagues.

As echoed by others in the original thread I feel that having the ability to do this is extremely useful functionality of FastAPI.

Thanks!

**Comment by ginomempin** [comment]
@arjwilliams It's not a duplicate of #9794 because the example code posted in that discussion thread is not the same one in the FastAPI docs. It was using a `Config.schema_extra` and `openapi_extra`, then the answer to that thread was to follow the new `examples=` style usage as documented in https://fastapi.tiangolo.com/tutorial/schema-extra-example/#body-with-multiple-examples..... which is what the example code on _this_ discussion thread is using, and that which isn't working as documented.

I can confirm that verbatim copy-pasting the code from the docs doesn't work since `0.99`:  
https://fastapi.tiangolo.com/tutorial/schema-extra-example/#body-with-multiple-examples 

- `fastapi<0.99` - OK
- `fastapi==0.99.0` - does not work
- `fastapi==0.99.1` - does not work
- `fastapi==0.100.0` - does not work
- `fastapi==0.100.1` - does not work

**Reply by arjwilliams** [reply]
Hi @ginomempin,

Although the examples given in each question are different, the underlying issue is the same - that FastAPI has moved the examples from the OpenAPI schema to the JSON schema, which Swagger UI doesn't (yet) fully support. Both the `schema_extra={'examples':...}` and the `Body(examples=...)` do the same thing to the generated `openapi.json` file.

And if you look into the generated `openapi.json` file of the example given in the docs, you can see that all of the examples are included in there (not just the first), which is why the answer in #9794 is saying that the issue is with Swagger UI to fix - FastAPI is passing a valid schema to it, with all of the examples, but it is just that they aren't rendering those examples as a dropdown when they are in the JSON schema.

Alternatively, the PR #9866 that is linked in #9794 is looking to address this on the FastAPI side by also adding the examples to the place in the `openapi.json` that would be compatible with Swagger UI, although in that PR you can also see under `Options for moving forward` that:

> It's entirely possible that [the move to OpenAPI 3.1.0] deliberately removed the functionality that enabled the multiple examples picker in Swagger UI, and if that's the case, let's just update the docs to remove the image that shows the dropdown picker.

so in either case, that PR would solve the issue of the example in the documentation being wrong.
